akka-actor/src/main/scala/akka/dispatch/BatchingExecutor.scala

```diff
@@ -78,22 +78,23 @@ private[akka] trait BatchingExecutor extends Executor {
     }
   }
 
+  private[this] val _blockContext = new ThreadLocal[BlockContext]()
+
   private[this] final class BlockableBatch extends AbstractBatch with BlockContext {
-    private var parentBlockContext: BlockContext = _
     // this method runs in the delegate ExecutionContext's thread
     override final def run(): Unit = {
       require(_tasksLocal.get eq null)
       _tasksLocal set this // Install ourselves as the current batch
-      val prevBlockContext = BlockContext.current
+      val prevBlockContext = _blockContext.get
+      _blockContext.set(BlockContext.current)
       BlockContext.withBlockContext(this) {
```

Comment:
Well, we cannot access the stack frame of `withBlockContext`, so we cannot use that knowledge here AFAICS. And yes, Scala needs to be fixed as well. BTW, why doesn’t the Scala version have the non-Blockable version as well?